### PR TITLE
Support basic create user statement

### PR DESCRIPTION
**Pull request by shenli**

### CODE REVIEW

File: parser/coldef/opt.go [hunk 1]
```diff
@@ -221,3 +221,18 @@ func (tc *TableConstraint) String() string {
 	}
 	return strings.Join(tokens, " ")
 }
+
+// AuthOption is used for parsing create use statement
```
Comment (qiuyesuifeng) [inline parser/coldef/opt.go]: End of .

File: parser/coldef/opt.go [hunk 2]
```diff
@@ -221,3 +221,18 @@ func (tc *TableConstraint) String() string {
...
+type AuthOption struct {
+	// AuthString/HashString can be empty, so we need to decide which one to use.
+	ByAuthString bool
+	AuthString   string
+	HashString   string
+	// TODO: support auth_plugin
+}
+
+// UserSpecification is used for parsing create use statement
```
Comment (qiuyesuifeng) [inline parser/coldef/opt.go]: End of .

File: stmt/stmts/account_manage.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+	"github.com/pingcap/tidb/model"
+	mysql "github.com/pingcap/tidb/mysqldef"
+	"github.com/pingcap/tidb/parser/coldef"
+	"github.com/pingcap/tidb/parser/opcode"
+	"github.com/pingcap/tidb/rset"
+	"github.com/pingcap/tidb/rset/rsets"
+	"github.com/pingcap/tidb/stmt"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/util/format"
+	"strings"
```
Comment (qiuyesuifeng) [inline stmt/stmts/account_manage.go]: It is better to clean import like:
origin package import
empty line
github.com package import

File: stmt/stmts/account_manage.go [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+	where := &rsets.WhereRset{
+		Src:  p,
+		Expr: composeUserTableFilter(name, host),
+	}
+	p, err = where.Plan(ctx)
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+	row, err := p.Next(ctx)
+	if err != nil {
```
Comment (qiuyesuifeng) [inline stmt/stmts/account_manage.go]: If there is an error, need close plan?

File: stmt/stmts/account_manage_test.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	. "github.com/pingcap/check"
+)
+
+func (s *testStmtSuite) TestCreateUserStmt(c *C) {
+	// Make sure user test not in mysql.User.
+	tx := mustBegin(c, s.testDB)
+	rows, err := tx.Query(`SELECT Password FROM mysql.User WHERE User="test" and Host="localhost"`)
+	c.Assert(err, IsNil)
+	rows.Next()
+	c.Assert(rows.Next(), IsFalse)
```
Comment (qiuyesuifeng) [inline stmt/stmts/account_manage_test.go]: Here why call rows.Next() twice?

**Comment by shenli** [comment]
PTAL @qiuyesuifeng

File: stmt/stmts/account_manage_test.go (hunk 5 above)
Comment (qiuyesuifeng) [inline stmt/stmts/account_manage_test.go]: I see also `rows.Next()` in line 39 and 62 and 76, need some update?

Comment (shenli) [inline stmt/stmts/account_manage_test.go]: No, those lines are useful. I use them to make sure there is only one row for certain condition.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**